File: 品味Java的21種設計模式/dp_cpp/builder/example2/ConcreteBuilder.cpp
#include "ConcreteBuilder.h"

namespace cn
{
	namespace javass
	{
		namespace dp
		{
			namespace builder
			{
				namespace example2
				{

					Product *ConcreteBuilder::getResult()
					{
						return resultProduct;
					}

					void ConcreteBuilder::buildPart()
					{
						//构建某个部件的功能处理
					}
				}
			}
		}
	}
}

File: 品味Java的21種設計模式/dp_java/builder/example2/ConcreteBuilder.java
package cn.javass.dp.builder.example2;
/**
 * 具体的构建器实现对象
 */
public class ConcreteBuilder implements Builder {
    /**
     * 构建器最终构建的产品对象
     */
    private Product resultProduct;
    /**
     * 获取构建器最终构建的产品对象
     * @return 构建器最终构建的产品对象
     */
    public Product getResult() {
        return resultProduct;
    }
    public void buildPart() {
        //构建某个部件的功能处理
    }
}